frontend/smart-kickers-game/src/App.js

```diff
@@ -31,13 +43,36 @@ function App() {
     });
   }
 
+  function handleEndGame() {
+    setToggleGameScore(!toggleGameScore);
+  }
+
   return (
     <>
       <h1>Smart Kickers</h1>
-      <GameResults blueScore={blueScore} whiteScore={whiteScore} />
-      <center>
-        <Button onClick={() => handleResetGame()}>Reset game</Button>
-      </center>
+      {toggleGameScore === false ? (
+        <>
+          {<GameResults blueScore={blueScore} whiteScore={whiteScore} />}{' '}
+          {
+            <>
+              <center className="game-ending-buttons">
+                <Button onClick={() => handleResetGame()}>Reset game</Button>
+                <br />
+                <Button
+                  onClick={() => {
+                    setFinalScores({ blue: blueScore, white: whiteScore });
+                    handleEndGame();
+                  }}
+                >
+                  End game
+                </Button>
+              </center>
+            </>
+          }{' '}
+        </>
+      ) : (
+        <GameStatistics finalScores={finalScores} handleEndGame={handleEndGame} handleResetGame={handleResetGame} />
```

Comment:
`finalScores={ {blue: blueScore, white: whiteScore} }`
and probably the rest of the code in `GameStatistics` should work as previously if you agree with my proposal you have to check that everything works as you want, but in my opinion, this new state is unnecessary.

Reply:
Unfortunately, this idea isn't working as we want.